### PR TITLE
dev_ping: don't crash with silly values.

### PR DESCRIPTION
**Pull request by rustyrussell**
It's a dev command, but still.

Fixes: #985

### CODE REVIEW

File: lightningd/dev_ping.c [hunk 1]
```diff
@@ -62,13 +62,38 @@ static void json_dev_ping(struct command *cmd,
...
+
 	if (!json_tok_number(buffer, pongbytestok, &pongbytes)) {
 		command_fail(cmd, "'%.*s' is not a valid number",
 			     (int)(pongbytestok->end - pongbytestok->start),
 			     buffer + pongbytestok->start);
 		return;
 	}
 
+	if (pongbytes > 65535) {
+		command_fail(cmd, "pongbytes only 16 bits, not %u", pongbytes);
```
Comment (ZmnSCPxj) [inline lightningd/dev_ping.c]: The message reads wonky?  Maybe better "pongbytes (%u) > 65535"?

Comment (ZmnSCPxj) [reply lightningd/dev_ping.c]: or "pongbytes must be <= 65535, is %u", or something...

File: lightningd/dev_ping.c [hunk 2]
```diff
@@ -62,13 +62,38 @@ static void json_dev_ping(struct command *cmd,
...
+	}
+
 	if (!json_tok_number(buffer, pongbytestok, &pongbytes)) {
 		command_fail(cmd, "'%.*s' is not a valid number",
 			     (int)(pongbytestok->end - pongbytestok->start),
 			     buffer + pongbytestok->start);
 		return;
 	}
 
+	if (pongbytes > 65535) {
```
Comment (cdecker) [inline lightningd/dev_ping.c]: The pong also has 2 byte `byteslen` field, so that should be `65535 - 2` I guess

Comment (rustyrussell) [reply lightningd/dev_ping.c]: Spec says values >= 65532 mean "no pong", so it's actually valid, and indeed, we test this.

File: lightningd/dev_ping.c (hunk 1 above)
Comment (rustyrussell) [reply lightningd/dev_ping.c]: Ack, fixed.

File: lightningd/dev_ping.c (hunk 2 above)
Comment (rustyrussell) [reply lightningd/dev_ping.c]: Note added.

**Comment by cdecker** [comment]
Rebased and waiting for CI to build before merging

ACK fcbc8be

**Pull request merged**